snuba/datasets/metrics_messages.py

```diff
@@ -25,6 +26,7 @@ class AggregationOption(Enum):
 INT_FLOAT_EXPECTED = "Int or Float expected"
 
 # These are the hardcoded values from the materialized view
+GRANULARITY_TEN_SECONDS = 0
```

Comment:
Is the retention going to stay 90 days for the 10 seconds granularity?
This should not be a valid product requirement.  The usefulness of 10 seconds aggregated data from 80 days ago is very questionable. While the cost is substantial. 
I would advise being able to tune the retention time in function of granularity. So the user can the retention time they want for each granularity. 
Preferably, we would limit the max retention for 10 seconds granularity.

Reply:
Yep, we talked about this for the upcoming product usage of 10s granularity. They will not be using 90 days retention. 

We can look into limiting the max retention in a subsequent PR so that we can unblock the team for now, if that sounds fair to you?

Reply:
I think we would be able to customize the retention based on granularity, after applying the new matviews (described in my comment below).

Reply:
Personally, I do question the value 10s granularity provides. It means the product cares about deviances which might happen within 10s boundary intervals. Which I am not sure is needed or not. If there was a sustained deviance then a 60s granularity would provide some of the same signals that a 10s granularity would provide but at a very substantial cheaper cost. There could even be some middle ground where we provide 20s or 30s granularity but at a cheaper cost. But I do not know how the cost model is going to work for the custom use cases so we can revisit this topic in the future.

> Also, we had discussed this in the product requirements and I believe we're all on the same page that 10s granularity shouldn't be stored at 90 day retention.
Do we know how long of a retention is needed for the 10s granularity?

Reply:
Yeah, I think it's valid to wonder what the exact need is for 10s granularity here. I think we can try to unblock the team for now, and maybe revisit this question once we all have a sense of the cost involved.

I'm double checking with the team on how much retention they need. 90 days was not a requirement, but we didn't talk about how long of a retention is required exactly.

Reply:
Yes, we can revisit the topic later. No need to block merging this PR because of it.